modules/standard/List.chpl

```diff
@@ -361,6 +361,12 @@ module List {
       ref result = array[itemIdx];
       return result;
     }
+    inline proc ref _getRef(idx: int) ref {
```

Comment:
In general, for cases like this where we have to "forcibly" tell the compiler that a method is ref, is it always enough to say that "if this method returns a ref, then the receiver is ref as well"? 

In my mind `ref` is used when the receiver is mutated inside the method, but I guess that's not entirely correct.

Reply:
I guess `this` indexing routines are a special case, though...